python/tk_multi_workfiles/file_save_form.py

```diff
@@ -181,7 +186,21 @@ def _on_use_next_available_version_toggled(self, checked):
 
     def _start_preview_update(self):
         """
+        Starts the path preview task if we're not initializing the gui. If a preview task is
+        currently running, it will be stopped and a new one will be launched.
         """
+
+        # When initializing the gui, events are fired multiple times due to signals
+        # emitted from the widgets. Here we are muting the start_preview_update call
+        # to avoid creating and deleting the preview task multiple times, which is not only
+        # wasteful but also makes the code harder to debug since 5 tasks end up computing
+        # the preview.
+        if self._initializing:
+            return
+
+        # Disable the button while the path is computed.
+        self._disable_save("Please wait while Toolkit calculates the next available file name.")
```

Comment:
> The only other thing to make it even more robust would be to perform a complete search (rather than using the cached result) when the user hits save.
Actually we are already not using the cached result by not asking for the dirty results (clean_only=True) in the _generate_path method. Or is there something else we're missing?